### PR TITLE
Remove size checks (unnecessary?). Fixes iOS.

### PR DESCRIPTION
**Pull request by xsacha**

### CODE REVIEW

**Comment by ruslo** [comment]
I've updated CI configs, waiting for tests to finish:
* https://travis-ci.org/hunter-packages/jpeg/builds/269587377?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> I've updated CI configs, waiting for tests to finish

Merged to branch `hunter`. Please rebase.

**Comment by xsacha** [comment]
Rebased

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/jpeg/releases/tag/v9b-p2

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,22 +64,6 @@ check_include_file(strings.h HAVE_STRINGS_H)
 if( HAVE_STRINGS_H AND NOT HAVE_STRING_H )
   set(NEED_BSD_STRINGS 1)
 endif()
-check_type_size(size_t SIZE_T) # -> HAVE_SIZE_T
-if( NOT HAVE_SIZE_T )
-  message(WARNING "Type size_t is not defined in any of the usual places.
-    Try putting 'typedef unsigned int size_t;' in jconfig.h."
-  )
-endif()
-check_type_size("unsigned char" UNSIGNED_CHAR) # -> HAVE_UNSIGNED_CHAR
```
Comment (ruslo) [inline CMakeLists.txt:74]: Used here: https://github.com/hunter-packages/jpeg/blob/019f51b1964216949a9b902c79fb4975cd3c98c6/jconfig-cmake.cfg#L5

Comment (xsacha) [reply CMakeLists.txt:74]: Doesn't that mean it is hardcoded to 1 and will become:
#define HAVE_UNSIGNED_CHAR 1

**Comment by xsacha** [comment]
Ok, this was a misunderstanding on my part. I hope I didn't do this elsewhere.
It became:
/* #undef HAVE_UNSIGNED_CHAR */
/* #undef HAVE_UNSIGNED_SHORT */
In include/jconfig.h

File: CMakeLists.txt (hunk 1 above)
Comment (ruslo) [reply CMakeLists.txt:74]: > Doesn't that mean it is hardcoded to 1 and will become

No, see how `configure_file` works:
* https://cmake.org/cmake/help/latest/command/configure_file.html

**Pull request merged**